File: c++/week1/day5.cpp
#include <iostream>

using namespace std;
void arrays(int *p, int size, int &max, int &sum){
    max = *p;
    for(int i = 0; i < size; i++){
        sum += *(p+i);
        if(*(p+i) > max){
            max = *(p+i);
        }
    }

    cout << "Sum : " << sum << endl;
    cout << "Max : " << max << endl;
}
int main(){
    int arrsize;
    int max;
    int sum;
    cout << "input array size" <<endl;
    cin >> arrsize;
    int arr[arrsize];
    for(int i = 0; i < arrsize; i++){
        if (i == 0)
            cout << "input 1st number" << endl;
        else if (i == 1)
            cout << "input 2st number" << endl;
        else
            cout << "input " << i+1 << "th number" << endl;

        cin >> arr[i];
    }
     arrays(arr,arrsize,max,sum);
    return 0;
}

File: python/week1/day5.py
'''문제 설명
정수로 이루어진 리스트가 주어진다.
리스트에 들어있는 짝수만 골라서 새로운 리스트를 만들어라
짝수들의 합과 최댓값을 출력하라
짝수가 하나도 없으면 "짝수가 없습니다"를 출력하라'''

nums = [3, 10, 7, 6, 1, 8, 9]
evens = []
sum = 0
max_even = None
for i in nums:
    if i % 2 == 0:
        evens.append(i)
        sum += i
        if max_even is None or i > max_even:
            max_even = i

print(f"Even list: {evens}")
print(f"Sum: {sum}")
print(f"Max Value: {max_even}")